src/image.js

```diff
@@ -1,18 +1,35 @@
-import { pipe, withConstructor, withStaticProperty } from '@flekschas/utils';
+import {
+  assign,
+  pipe,
+  withConstructor,
+  withStaticProperty
+} from '@flekschas/utils';
 import * as PIXI from 'pixi.js';
 
 import withColorFilters from './with-color-filters';
+import withDestroy from './with-destroy';
 import withSize from './with-size';
 
 const createImage = (texture, { anchor = [0.5, 0.5] } = {}) => {
-  const sprite = new PIXI.Sprite(texture);
+  let sprite = new PIXI.Sprite(texture);
   sprite.anchor.set(...anchor);
 
+  const updateSprite = newTexture => {
+    sprite = new PIXI.Sprite(newTexture);
+  };
+
```

Comment:
If I update the sprite like this, then call `image.updateSprite()` outside, the sprite is actually not updated. Does it has something to do with the `withStaticProperty` so the property is static, or did I do something wrong?

Reply:
That might be the case yes. An easy way to test this is to replace:
```
withStaticProperty('displayObject', sprite),
withStaticProperty('sprite', sprite),
```
with
```
withReadOnlyProperty('displayObject', () => sprite),
withReadOnlyProperty('sprite', () => sprite),
```
One other small thing: can we just call the method `updateTexture` or even just `update`. `updateSprite` is a bit confusing because you pass in a texture rather than a sprite object.

Reply:
Also, one thing we need to do is to destroy the previous sprite to free up its memory. So _before_ you overwrite the sprite, make sure to call `sprite.destroy()`!

Reply:
Now I am thinking there might also be issues with `withColorFilters(sprite)`, `withSize(sprite)`, and `withDestroy(sprite)` as they are all linked to the old sprite object.

A cleaner solution for updating the image might be to not update it at all and instead destroy the old image and replace it with a new image on the pile-item. We could add a method there called `replaceImage(newImage, newPreview)`. In that method we'd first have to `.destroy()` the old image and preview and remove them from the root PIXI container then just add the new image with `addChildAt(newImage, oldIndex)`.

The more I think about it the more I like this solution. Can you try to implement it this way? Thanks :)